angular/server.ts

```diff
@@ -0,0 +1,57 @@
+import { APP_BASE_HREF } from '@angular/common';
+import { CommonEngine } from '@angular/ssr';
+import fastifyStatic from '@fastify/static';
+import fastify, { FastifyInstance } from 'fastify';
+import { dirname, join, resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import bootstrap from './src/main.server';
+
+export function app(): FastifyInstance {
+  const server = fastify();
+  const serverDistFolder = dirname(fileURLToPath(import.meta.url));
+  const browserDistFolder = resolve(serverDistFolder, '../browser');
+  const indexHtml = join(serverDistFolder, 'index.server.html');
+
+  const commonEngine = new CommonEngine();
+
+  server.register(fastifyStatic, {
+    root: browserDistFolder,
+    maxAge: '1y',
+    index: false,
+    wildcard: false,
+  });
+
+  server.get('*', (req, res) => {
+    const { protocol, url, hostname } = req;
+    const baseUrl = protocol + '://' + hostname;
+
+    return commonEngine
+      .render({
```

Comment:
If your point to to test SSR only, you may want to disable critical CSS inlining here.

https://angular.dev/guide/ssr#configure-server-side-rendering

Reply:
Thanks for the feedback, this improved a bit the results (see next comment)